### PR TITLE
11847 error handling school information fetching

### PR DESCRIPTION
**Pull request by rianfowler**

### CODE REVIEW

File: src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js [hunk 1]
```diff
@@ -42,7 +42,34 @@ export default function schoolSearch(state = initialState, action) {
...
+    case LOAD_SCHOOLS_FAILED: {
+      const currentPageNumber = 0;
+      const institutionSelected = {};
+      const institutionQuery = action.institutionQuery;
+      const institutions = [];
+      const pagesCount = 0;
+      const searchResultsCount = 0;
+      const showInstitutions = false;
+      const showInstitutionsLoading = false;
+      const showNoResultsFound = true;
```
Comment (rianfowler) [inline src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js]: Handle school loading failure by showing no results found.

Comment (cehsu) [reply src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js]: Should we distinguish no results found from unable to retrieve results due to an error? Perhaps a message such as "search is unavailable at this time, please try again later or use the manual entry fields below" and log the errors?

File: src/applications/edu-benefits/complaint-tool/helpers.js [hunk 2]
```diff
@@ -15,11 +15,18 @@ export function fetchInstitutions({ institutionQuery, page }) {
       'X-Key-Inflection': 'camel'
     }
   })
-    .then(res => res.json())
+    .then(res => {
```
Comment (cehsu) [inline src/applications/edu-benefits/complaint-tool/helpers.js]: could we use the [apiRequest helper ](https://github.com/department-of-veterans-affairs/vets-website/blob/dc5cd8642a10aa654a9fe54b53d55dccafce6acc/src/platform/utilities/api/index.js#L13)here?

**Review by cehsu — APPROVED** [review]
LGTM, some questions/comments for your consideration.

File: src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js (hunk 1 above)
Comment (rianfowler) [reply src/applications/edu-benefits/complaint-tool/reducers/schoolSelect.js]: I'll make a ticket for this.

File: src/applications/edu-benefits/complaint-tool/helpers.js (hunk 2 above)
Comment (rianfowler) [reply src/applications/edu-benefits/complaint-tool/helpers.js]: I'm pretty against using fetch abstractions. This one for example does a redirect if the response status is 401. 😱 That is a really invasive + obfuscated behavior.

**Pull request merged**